rust-crates/symblib-capi/src/pointresolver.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{StatusCode, SymblibSlice, SymblibString};
+use symblib::symbconv;
+use symblib::symbconv::PointResolver;
+use symblib::VirtAddr;
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct SymblibResolvedSymbol {
+    pub start_addr: VirtAddr,
+    pub function_name: SymblibString,
+    pub file_names: SymblibSlice<SymblibString>,
+    pub line_numbers: SymblibSlice<u32>,
+}
+
+impl From<symbconv::ResolvedSymbol> for SymblibResolvedSymbol {
+    fn from(sym: symbconv::ResolvedSymbol) -> Self {
+        let file_names: Vec<SymblibString> = sym
+            .file_names
+            .unwrap_or_default()
+            .into_iter()
+            .map(Into::into)
+            .collect();
+
+        let line_numbers: Vec<u32> = sym.line_numbers.unwrap_or_default();
+
+        Self {
+            start_addr: sym.start_addr,
+            function_name: sym.function_name.into(),
+            file_names: file_names.into(),
+            line_numbers: line_numbers.into(),
+        }
+    }
+}
+
+#[repr(C)]
+pub struct SymblibPointResolver {
```

Comment:
An even more explicit lifetime option would be:
```
pub struct SymblibPointResolver {
    inner: Box<dyn PointResolver + Send + 'static>,
}
```
`SymblibGoRuntime` already meets this additional `'static` requirement, so it could be fine to use. Let me know if you would prefer this option.

Reply:
`'static` is default unless otherwise specified.